### ISSUE TITLE
Readme Unclear

### ISSUE THREAD

**Issue by darinkishore** [shroominic/codeinterpreter-api#2]
What does "You can run everything local using your own OpenAI API Key" mean in the readme? We can execute code locally just by providing an openai api key? That doesn't really seem to make sense, would love some clarification!

Thank you for publishing the project--code interpreter usage is easily the most powerful I've seen GPT get, so I'm super excited about the possibilities of this.

**Comment by shroominic** [comment]
Thank you for pointing this out. Let me try to make it clearer.

When I said, "You can run everything local using your own OpenAI API Key," I was referring to the Python code execution part of the project. The code interpreter is basically ChatGPT with a Python Interpreter as tool to use in helping the user.

However, please note that the LLM, be it GPT-3.5 or GPT-4, doesn't run locally. Since OpenAI hasn't yet open-sourced these models, requests for their usage still need to be sent via their API.

So, to sum up, all Python code execution and the Langchain Agent logic surrounding the OpenAI API are run locally, but GPT-4 itself is not.

**Comment by darinkishore** [comment]
Thanks, great clarification. Makes a lot more sense now.

Question—Love the idea and premise of the project. I can’t seem to see how dialog would work in it, though. If it’s just single completion-responses hardcoded in a script, can it still be interactive? Does that make sense?

**Comment by shroominic** [comment]
Yeah I should make an example for this but you can just request a follow up response to the same session.
